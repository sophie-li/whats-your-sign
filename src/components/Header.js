import React, { Component } from 'react';

class Header extends Component {
  shouldComponentUpdate(nextProps, nextState) {
    return false;
  }

  render() {
    return (
      <div>
        <h1 className='f1'>WHAT'S YOUR SIGN?</h1>
      </div>
    );
  }
}

export default Header;
