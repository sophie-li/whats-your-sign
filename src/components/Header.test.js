import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Header />);
});

it('expect to render Header component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should not update component', () => {
  const shouldUpdate = wrapper
    .instance()
    .shouldComponentUpdate({ props: 'bla' }, {});
  expect(shouldUpdate).toBe(false);
});
