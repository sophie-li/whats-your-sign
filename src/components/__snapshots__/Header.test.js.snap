// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `
<h1
  className="f1"
>
  WHAT'S YOUR SIGN?
</h1>
`;
